### ISSUE TITLE
MSAL crashes when the passed authority uses domain suffix instead of guid.

### ISSUE THREAD

**Issue by rpdome** [AzureAD/microsoft-authentication-library-for-android#704]
**Scenario**:
The customer is trying to pass the non-guid authority, (i.e. https://login.microsoftonline.com/testtenant.onmicrosoft.com) as the optional authority of the acquireTokenSilent call, and the app crashes.

**Root cause**:
In OperationParametersAdapter.createAcquireTokenSilentOperationParameters(), we have a check to decide if we want to use home or tenant profile. In this case, however, isAccountHomeTenant failed because that function was expecting 'tenantId' to be GUID, but was getting "testtenant.onmicrosoft.com" instead. Therefore, this function returns false and we're trying to use tenant profile, which was null and caused the crash.

**Comment by iambmelt** [comment]
Note: Testing this fix must also verify B2C is working (incl. w/ `b2clogin.com` style urls)

/cc @negoe

**Comment by iambmelt** [comment]
```
    --------- beginning of crash
2019-08-16 09:50:55.198 31997-31997/com.microsoft.identity.client.sample.local E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.microsoft.identity.client.sample.local, PID: 31997
    java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Map com.microsoft.identity.client.ITenantProfile.getClaims()' on a null object reference
        at com.microsoft.identity.client.internal.controllers.OperationParametersAdapter.createAcquireTokenSilentOperationParameters(OperationParametersAdapter.java:292)
        at com.microsoft.identity.client.PublicClientApplication.acquireTokenSilentAsync(PublicClientApplication.java:1106)
        at com.microsoft.identity.client.testapp.MsalWrapper$3.onTaskCompleted(MsalWrapper.java:264)
        at com.microsoft.identity.client.MultipleAccountPublicClientApplication$4.onTaskCompleted(MultipleAccountPublicClientApplication.java:295)
        at com.microsoft.identity.client.MultipleAccountPublicClientApplication$4.onTaskCompleted(MultipleAccountPublicClientApplication.java:266)
        at com.microsoft.identity.common.internal.controllers.ApiDispatcher$1$2.run(ApiDispatcher.java:101)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by iambmelt** [comment]
Fix in progress, this requires querying `.well-known` config

**Comment by iambmelt** [comment]
Resolved by #710
